cmd/juju/application/deployer/lxdprofilestep.go

```diff
@@ -4,49 +4,10 @@
 package deployer
 
 import (
-	"github.com/go-macaroon-bakery/macaroon-bakery/v3/httpbakery"
 	"github.com/juju/charm/v10"
-	"github.com/juju/cmd/v3"
-	"github.com/juju/errors"
-	"github.com/juju/gnuflag"
-
-	apicharms "github.com/juju/juju/api/common/charms"
 	"github.com/juju/juju/core/lxdprofile"
 )
 
-// ValidateLXDProfileCharm implements the DeployStep interface.
-type ValidateLXDProfileCharm struct{}
-
-// SetFlags implements DeployStep.
-func (r *ValidateLXDProfileCharm) SetFlags(f *gnuflag.FlagSet) {
-}
-
-// RunPre obtains authorization to deploy this charm. The authorization, if received is not
-// sent to the controller, rather it is kept as an attribute on RegisterMeteredCharm.
-func (r *ValidateLXDProfileCharm) RunPre(api DeployStepAPI, bakeryClient *httpbakery.Client, ctx *cmd.Context, deployInfo DeploymentInfo) error {
-	// if the charm info is not empty, we should use that to validate the
-	// lxd profile.
-	if charmInfo := deployInfo.CharmInfo; charmInfo != nil {
-		if err := lxdprofile.ValidateLXDProfile(lxdCharmInfoProfiler{
-			CharmInfo: charmInfo,
-		}); err != nil {
-			// The force flag was provided, but we should let the user know that
-			// this could deliver some unexpected results.
-			if deployInfo.Force {
-				logger.Debugf("force option used to override validation error %v", err)
-				return nil
-			}
-			return errors.Trace(err)
-		}
-	}
-	return nil
-}
-
-// RunPost sends credentials obtained during the call to RunPre to the controller.
-func (r *ValidateLXDProfileCharm) RunPost(api DeployStepAPI, bakeryClient *httpbakery.Client, ctx *cmd.Context, deployInfo DeploymentInfo, prevErr error) error {
-	return nil
-}
-
 // lxdCharmProfiler massages a charm.Charm into a LXDProfiler inside of the
 // core package.
 type lxdCharmProfiler struct {
```

Comment:
renamed the file if we're keeping this, a bit confusing without the deploy steps